addons/marketing_card/wizards/mail_compose_message.py

```diff
@@ -0,0 +1,52 @@
+import re
+
+from odoo import api, models
+
+CARD_IMAGE_URL = re.compile(r'src=".*?/web/image/card.campaign/[0-9]+/image_preview"')
+CARD_PREVIEW_URL = re.compile(r'href=".*?/cards/[0-9]+/preview"')
+
+
+class MailComposeMessage(models.TransientModel):
+    _inherit = 'mail.compose.message'
+
+    @api.model
+    def _process_generic_card_url_body(self, card_body_pairs: list[tuple[models.Model, str]]) -> list[str]:
```

Comment:
:socks: would define after _prepare_mail_values (I know there is a debate, my pov is that you read from top to bottom so first main method, then method called in it, ...).

Reply:
That's fine, I guess out of habit it's "safer" to put them before since you'd need to outside of classes but :shrug: potato potato

Reply:
Oh and don't forget typing annotations are deprecated since v18.